dask/dataframe/core.py

```diff
@@ -1654,6 +1661,44 @@ def count(self, axis=None, split_every=False):
                 result.divisions = (self.columns.min(), self.columns.max())
             return result
 
+    @derived_from(pd.DataFrame)
+    def mode(self, axis=0, dropna=True, split_every=False):  # this is the series class
+        """Use a counter on each partition, sum the resulting counters, iterate over key,
+        value pairs until num_occurrences isn't equal to the first one"""
+        if axis == 1:
```

Comment:
Users can also pass `axis='columns'` You should normalize the axis with `axis = self._validate_axis(axis)`.

Reply:
The axis are validated in the DataFrame mode method which calls this method so it wouldn't be necessary to _validate_axis again.